rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMPage.java

```diff
@@ -425,25 +420,12 @@ public void onClick(MenuItemClickEvent event) {
         ToolStripButton nsButton = new ToolStripButton("Add Node Source");
         nsButton.setIcon(RMImages.instance.nodesource_deployed_16().getSafeUri().asString());
         nsButton.setTooltip("Create and add a new Node Source");
-        nsButton.addClickHandler(new ClickHandler() {
-            public void onClick(ClickEvent event) {
-                if (RMPage.this.nsWindow != null)
-                    RMPage.this.nsWindow.destroy();
-                RMPage.this.nsWindow = new NSCreationWindow(controller);
-                RMPage.this.nsWindow.show();
-            }
-        });
+        nsButton.addClickHandler(e -> showNodeSourceCreationWindow());
 
         errorButton = new ToolStripButton("<strong>Error</strong>",
                                           Images.instance.net_error_16().getSafeUri().asString());
         errorButton.setBackgroundColor("#ffbbbb");
-        errorButton.addClickHandler(new ClickHandler() {
-            @Override
-            public void onClick(ClickEvent event) {
-                RMPage.this.logWindow.show();
-                errorButton.hide();
-            }
-        });
+        errorButton.addClickHandler(e -> showLogWindow());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)